src/sql/src/session/vars.rs

```diff
@@ -1361,6 +1385,9 @@ impl SystemVars {
 
     /// Returns whether or not this parameter can be modified by a superuser.
     pub fn user_modifiable(&self, name: &str) -> bool {
+        if self.check_read_only(name).is_err() {
+            return false;
+        }
```

Comment:
MAX_CREDIT_CONSUMPTION_RATE shouldn't be a session_var so it shouldn't need a readonly check here, although it's not hurting anything to be here.

Reply:
ah, fair enough, but yeah, i think it still makes sense to do this to make it easier to understand in the future